sdk/eventhub/Azure.Messaging.EventHubs/assets/identity-samples-azure-setup.ps1

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+  .SYNOPSIS
+    This script handles creation of the principal and the resources needed to run identity samples.
+
+  .DESCRIPTION
+    It tries to retrieve a resource group, an Azure Event Hubs Namespace and an Azure Event Hub
+    using the names passed in as arguments. It attempts the creation of the resources if not found.
+
+    It always tries to create the named service principal.
+    
+    It assigns the 'Azure Event Hubs Data Owner' role to the namespace.
+
+    Upon completion, the script will output the principal's sensitive information and the parameters
+    needed to run the samples.
+ 
+    For more detailed help, please use the -Help switch. 
+#>
+
+# =======================
+# == Script Parameters ==
+# =======================
+
+[CmdletBinding(DefaultParameterSetName="Help")] 
+[OutputType([String])] 
+param
+( 
+  [Parameter(Mandatory=$true, ParameterSetName="Help", Position=0)]
+  [Switch] $Help,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute", Position=0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $SubscriptionName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [string] $ResourceGroupName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [string] $NamespaceName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [string] $EventHubName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [ValidateScript({ $_.Length -ge 6})]
+  [string] $ServicePrincipalName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [AllowNull()]  
+  [string] $AzureRegion = $null
+)
+
+# =====================
+# == Module Imports  ==
+# =====================
+
+Import-Module Az.Resources
+
+# ==========================
+# == Function Definitions ==
+# ==========================
+
+. .\functions\validation-functions.ps1
+. .\functions\help-functions.ps1
+. .\functions\credentials-functions.ps1
```

Comment:
I will say this.... the script is very readable and approachable at a high-level with the details tucked away in the imports.